api/views/student.py

```diff
@@ -105,6 +109,34 @@ def missed_class(self, request: Request, personal_info_id: int) -> Response:  #
         )
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @extend_schema(
+        parameters=[DashboardAvailableStudentsSerializer],
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(
+                response=ListSerializer(child=MinifiedStudentSerializer()),
+                description="Action is taken",
+            ),
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                response=ValidationErrorSerializer,
+                description="Something is wrong with the query params",
+            ),
+        },
+    )
+    @action(detail=False, methods=["get"])
+    def available_students_list(self, request: Request) -> Response:  # noqa: ARG002
+        query_params_serializer = DashboardAvailableStudentsSerializer(
+            data=request.query_params,
+        )
+        query_params_serializer.is_valid(raise_exception=True)
+        matched_students = Student.objects.filter(
```

Comment:
I know it is bad to have it here but I though that creating a separate processor might be too much, if you have ideas to where exactly put it feel free to tell